libraries/shared/storage/fetcher/geth_rpc_storage_fetcher.go

```diff
@@ -17,28 +17,23 @@ package fetcher
 import (
 	"fmt"
 	"io/ioutil"
+	"math/big"
 
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/eth/filters"
 	"github.com/ethereum/go-ethereum/rlp"
-	"github.com/ethereum/go-ethereum/statediff"
 	"github.com/makerdao/vulcanizedb/libraries/shared/storage/types"
 	"github.com/makerdao/vulcanizedb/libraries/shared/streamer"
 	"github.com/sirupsen/logrus"
 )
 
-type GethPatchVersion int
-
-const (
-	OldGethPatch GethPatchVersion = iota
-	NewGethPatch
-)
-
 type GethRpcStorageFetcher struct {
-	statediffPayloadChan chan statediff.Payload
+	statediffPayloadChan chan filters.Payload
 	streamer             streamer.Streamer
 	gethVersion          GethPatchVersion
 }
 
-func NewGethRpcStorageFetcher(streamer streamer.Streamer, statediffPayloadChan chan statediff.Payload, gethVersion GethPatchVersion) GethRpcStorageFetcher {
+func NewGethRpcStorageFetcher(streamer streamer.Streamer, statediffPayloadChan chan filters.Payload, gethVersion GethPatchVersion) GethRpcStorageFetcher {
```

Comment:
Not a blocker but why are we passing the `statediffPayloadChan` in the constructor as opposed to in the call to `FetchStorageDiffs` and/or just initializing it on demand?

Reply:
The only reason I can think of why we're passing `statediffPayloadChan` in is to make it easier to test. That being said, I think we'd get the same outcome by passing it in to `FetchStorageDiffs` instead of to the constructor.